File: 백준/Gold/1644. 소수의 연속합/소수의 연속합.py
import sys
input = sys.stdin.readline
import math

n = int(input())
##print(n)
isPrime = [False, False] + [True]*(n-1)
seqPrime = []

for i in range(2,n+1):
    if isPrime[i]:
        seqPrime.append(i)
        for j in range(2*i, n+1, i):
            isPrime[j]=False
            

start = end = 0
total = 0
answer = 0
while True:
    if total>=n:
        total-=seqPrime[start]
        start+=1
    elif end==len(seqPrime):
        break
    else:
        total+=seqPrime[end]
        end+=1
    if total==n:
        answer+=1

    ##print(total, start, seqPrime[start], end, seqPrime[end])
print(answer)
    

File: 백준/Gold/1644. 소수의 연속합/소수의 연속합.js
const readline = require("readline")
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
})
const lines = []
rl.on("line", (line) => {
    lines.push(line.trim())
}).on(("close"), () => {

    const N = Number(lines[0])
    const lst = new Array(N + 1).fill(true)
    lst[0] = lst[1] = false
    for (let i = 2; i < N + 1; i++) {
        for (let j = i + i; j < N + 1; j += i) {
            lst[j] = false

        }



    }
    const arr = lst.map((e, idx) => (e === true ? idx : 0)).filter((e) => e != 0)
    let start = 0
    let end = start
    let total = arr[end]
let answer = 0

    while (start<=end &&end < N + 1) {
        if (total <= N) {
            if (total == N) {
                // console.log(arr.slice(start, end + 1))
answer+=1

            }
            end += 1
total+=arr[end]

        } else {
total-=arr[start]

            start += 1
        }

    }

    console.log( answer)
})